business/workloads.go

```diff
@@ -989,3 +992,52 @@ func controllerPriority(type1, type2 string) string {
 		return type2
 	}
 }
+
+func (in *WorkloadService) fillCustomDashboardRefs(namespace string, workload *models.Workload) {
+	workload.CustomDashboards = []models.CustomDashboardRef{}
+	uniqueRefs := make(map[string]string)
+	for _, pod := range workload.Pods {
+		for _, ref := range pod.CustomDashboards {
+			if ref != "" {
+				uniqueRefs[ref] = ref
+			}
+		}
+	}
+	mon, err := kubernetes.NewKialiMonitoringClient()
+	if err != nil {
+		log.Error("Cannot get dashboards - cannot load Kiali Monitoring Client")
+		return
+	}
+	for _, ref := range uniqueRefs {
+		title := getCustomDashboardTitle(mon, namespace, ref)
+		if title != "" {
+			workload.CustomDashboards = append(workload.CustomDashboards, models.CustomDashboardRef{
+				ID:    ref,
+				Title: title,
+			})
+		}
+	}
+}
+
+// Memoize titles
+var customDashboardTitle = make(map[string]string)
+
+// TODO: put this function in DashboardsService when related PR in merged
+func getCustomDashboardTitle(mon *kubernetes.KialiMonitoringClient, namespace, name string) string {
+	// @ is forbidden charatecter in k8s resource name, so safe to use here
+	key := name + "@" + namespace
+	if title, ok := customDashboardTitle[key]; ok {
+		return title
+	}
+	dashboard, err := mon.GetDashboard(namespace, name)
```

Comment:
Similar comments here than in the apps.go code

Reply:
wrong commit